### PR TITLE
:sparkles: spawnVehicle Server

### PR DESCRIPTION
**Pull request by TheSpaceGamerV2**
This code was taken from here :
https://github.com/esx-framework/esx-legacy/blob/main/%5Besx%5D/es_extended/server/onesync.lua

There is no better way to do it, sometime peds can spawn in vehicles so we check if a ped is in the vehicle and if it's the case we delete him / them.

the 
```lua
repeat
    until
```
is here to wait until the vehicle is loaded on client side (can take time, with modded vehicle for example)

### CODE REVIEW

File: src/api/vehicles/server/spawnVehicle.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            ped = GetPedInVehicleSeat(vehicle, i)
+            local populationType = GetEntityPopulationType(ped)
+            if (populationType <= 5 or populationType >= 1) then
+                DeleteEntity(ped)
+            end
+        end
+    end
+
+    cb(vehicle, NetworkGetNetworkIdFromEntity(vehicle))
+end
```
Comment (PABLO-1610) [inline src/api/vehicles/server/spawnVehicle.lua:26]:
```suggestion
---spawn
---@param modelName string
---@param coords table (vector4)
---@param cb function
---@public
function API_Vehicles:spawn(modelName, coords, cb)
    local vehicle = Citizen.InvokeNative(`CREATE_AUTOMOBILE`, joaat(modelName), coords)
    repeat
        Wait(0)
    until DoesEntityExist(vehicle)

    -- Peds can spawn in vehicles (don't ask me why)
    local ped = GetPedInVehicleSeat(vehicle, -1)
    if (ped > 0) then
        for i = -1, 6 do
            ped = GetPedInVehicleSeat(vehicle, i)
            local populationType = GetEntityPopulationType(ped)
            if (populationType <= 5 or populationType >= 1) then
                DeleteEntity(ped)
            end
        end
    end

    cb(vehicle, NetworkGetNetworkIdFromEntity(vehicle))
end
```

**Review by PABLO-1610 — CHANGES_REQUESTED** [review]

**Comment by TheSpaceGamerV2** [comment]
Should be good now @PABLO-1610

File: src/api/vehicles/server/spawnVehicle.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+
```
Comment (PABLO-1610) [inline src/api/vehicles/server/spawnVehicle.lua:2]:
```suggestion
---serverSpawn
```

File: src/api/vehicles/server/spawnVehicle.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+---@param modelName string
+---@param coords table (vector4)
+---@param cb function
+---@public
+function API_Vehicles:spawn(modelName, coords, cb)
```
Comment (PABLO-1610) [inline src/api/vehicles/server/spawnVehicle.lua:7]:
```suggestion
function API_Vehicles:serverSpawn(modelName, coords, cb)
```

**Review by PABLO-1610 — CHANGES_REQUESTED** [review]

**Comment by PABLO-1610** [comment]
I think it's better to name this function `serverSpawn` since auto-completion will maybe show the API_Vehicles#spawn clientside parameters.

**Comment by TheSpaceGamerV2** [comment]
> I think it's better to name this function `serverSpawn` since auto-completion will maybe show the API_Vehicles#spawn clientside parameters.

Yeah u right, unfortunately we can't really detect if someone is in a client file or server file

File: src/api/vehicles/server/spawnVehicle.lua [hunk 4]
```diff
@@ -0,0 +1,26 @@
+--serverSpawn
+
```
Comment (PABLO-1610) [inline src/api/vehicles/server/spawnVehicle.lua:2]:
```suggestion
```
Ew 🤣

**Review by PABLO-1610 — CHANGES_REQUESTED** [review]

File: src/api/vehicles/server/spawnVehicle.lua (hunk 4 above)
Comment (TheSpaceGamerV2) [reply src/api/vehicles/server/spawnVehicle.lua:2]: Oops

File: src/api/vehicles/server/spawnVehicle.lua [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---serverSpawn
+
```
Comment (PABLO-1610) [inline src/api/vehicles/server/spawnVehicle.lua:2]:
```suggestion
```

**Review by PABLO-1610 — CHANGES_REQUESTED** [review]

File: src/api/vehicles/server/spawnVehicle.lua (hunk 5 above)
Comment (TheSpaceGamerV2) [reply src/api/vehicles/server/spawnVehicle.lua]: good now ? :sob:

Comment (PABLO-1610) [reply src/api/vehicles/server/spawnVehicle.lua]: > good now ? 😭

Yes 🙏

**Review by PABLO-1610 — APPROVED** [review]

**Pull request merged**